.github/workflows/fill_station_build_and_push.yml

```diff
@@ -0,0 +1,112 @@
+name: Fill Station Build Test Push
+
+# Configures this workflow to run every time a change is pushed to the branch called `main` or a pull request to `main` is created
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    branches: ['main']
+
+# Defines two custom environment variables for the workflow. These are used for the Container registry domain, and a name for the Docker image that this workflow builds.
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository_owner }}/fill-station
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, which is the CWD for
+      # the rest of the steps
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - uses: bazel-contrib/setup-bazel@0.8.5
+        with:
+          # Avoid downloading Bazel every time.
+          bazelisk-cache: true
+          # Store build cache per workflow.
+          disk-cache: ${{ github.workflow }}
+          # Share repository cache between workflows.
+          repository-cache: true
+
+        # test
+      - name: Build and Test with Bazel
+        run: bazel --output_base=/tmp/bazel/output test //fill:all
+
+        # Upload the fill station binary as an artifact
+      - uses: actions/upload-artifact@v4
+        with:
+          # Name of the artifact to upload.
+          # Optional. Default is 'artifact'
+          name: fill_station
+          # The path to the fill station binary from bazel build
+          path: /tmp/bazel/output/execroot/_main/bazel-out/k8-fastbuild/bin/fill/fill_station
```

Comment:
Do we wanna upload the executable as an artifact? Because it will only be runnable on x64 linux, which is not many computers. 

If we do want to upload it, I think we should suffix it with `linux-x86_64` or something similar. Also, is it possible for the path is relative instead of absolute so that it is not referencing a /tmp directory?

Reply:
The artifact is just so docker build and push can use it. We should never have to run the artifact itself

Reply:
I can put it in a different directory if we don't like tmp, but a relative path would still reference tmp. In general, we should prefer absolute paths, right?